test/LTO/pipeline.swift

```diff
@@ -0,0 +1,8 @@
+// RUN: %empty-directory(%t)
+// RUN: cd %t && %target-swiftc_driver -emit-module %S/Inputs/lib.swift
+// RUN: cd %t && %target-swift-frontend -emit-sib %S/Inputs/lib.swift -parse-as-library
+// RUN: cd %t && %target-swift-frontend -emit-sib -I%t  %S/Inputs/main.swift
+
+// Examine loading order
+// RUN: cd %t && %swift-lto main.sib lib.sib
+// RUN: cd %t && %swift-lto lib.sib main.sib
```

Comment:
This test doesn't actually check anything ... what is the purpose of the test?

Reply:
This test checks that modules can be deserialized using the pipeline and ensure that the pipeline can resolve dependent modules regardless of input order.

Reply:
I think Im not understanding something then.  The test doesn't verify that the module are deserialized and the dependency modules are resolved.  It just assumes that if the process exited everything is correct.

Reply:
I see. added more checks